tests/test_parser.py

```diff
@@ -1641,6 +1641,9 @@ def test_uncaptured_sub():
 def test_hiddenobj_sub():
     check_xonsh_ast({}, '![ls]', False)
 
+def test_slash_envarv_echo():
+    check_xonsh_ast({}, '![echo $HOME/place]', False)
```

Comment:
When `run=False`, as is passed in here, this checks that it is parseable and that the AST can be compiled into bytecode. It skips the actual execution.  In this case, since the original issue was actually a parser bug, I think this is enough.